pkg/vgmanager/vgmanager_controller.go

```diff
@@ -243,3 +284,32 @@ func ToleratesTaints(tolerations []corev1.Toleration, taints []corev1.Taint) boo
 	}
 	return true
 }
+
+func (r *VGReconciler) updateStatus(ctx context.Context, status string, reason string, instance *lvmv1alpha1.LVMVolumeGroup) error {
+
+	found := false
```

Comment:
We still have a situation where multiple controllers will be updating the same `LVMVolumeGroup` status, right?  
If that's the case, should we first get the latest `LVMVolumeGroup` instance before verifying the `NodeStatus`?

Reply:
I think this maybe be enough, to ensure that we are using the updated instance of `LVMVolumeGroup`.

Reply:
If it is not the latest version, I expect the update to fail. I will confirm after testing on a multi node cluster.